### PR TITLE
[BAEL-4268]Different Serialization Approaches for Java

### PR DESCRIPTION
**Pull request by MeenaGawande**
Code Changes

### CODE REVIEW

File: libraries-data-io/pom.xml [hunk 1]
```diff
@@ -61,6 +61,52 @@
             <version>${org.assertj.core.version}</version>
             <scope>test</scope>
         </dependency>
+         <!-- gson api -->
```
Comment (KevinGilmore) [inline libraries-data-io/pom.xml:64]: There's a mixture of indentation spaces and tabs. Let's apply spaces across-the-board to be like the rest of the file.

File: libraries-data-io/src/test/java/com/baeldung/serialization/protocols/ApacheThriftSerializationUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        TMemoryBuffer trans = new TMemoryBuffer(4096);
+        TProtocol proto = new TBinaryProtocol(trans);
+        proto.writeI32(user.getId());
+
+        proto.writeString(user.getName());
+        int userId = proto.readI32();
+
+        String userName = proto.readString();
+        assertEquals(2, userId);
+        assertEquals("Greg", userName);
```
Comment (KevinGilmore) [inline libraries-data-io/src/test/java/com/baeldung/serialization/protocols/ApacheThriftSerializationUnitTest.java:29]: Adjust placement of blank lines. I would put both the writes together, followed by a blank line, the the two reads followed by a blank line, and then the assertions.

File: libraries-data-io/src/test/java/com/baeldung/serialization/protocols/GoogleProtocolBufferUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    public void whenUsingProtocolBuffersSerialization_ThenObjectIsTheSameAfterDeserialization() throws IOException {
+
+        String filePath = "src/test/resources/protocols/usersproto";
+
+        UserProtos.User user = UserProtos.User.newBuilder().setId(1234).setName("John Doe").build();
+        FileOutputStream fos = new FileOutputStream(filePath);
+        user.writeTo(fos);
+
+        UserProtos.User deserializedUser = UserProtos.User.newBuilder().mergeFrom(new FileInputStream(filePath)).build();
+        assertEquals(1234, deserializedUser.getId());
```
Comment (KevinGilmore) [inline libraries-data-io/src/test/java/com/baeldung/serialization/protocols/GoogleProtocolBufferUnitTest.java:23]: Add a blank line before beginning the assertions.

File: libraries-data-io/src/test/java/com/baeldung/serialization/protocols/GsonSerializationUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        User user = new User();
+        user.setId(1);
+        user.setName("Mark");
+
+        String filePath = "src/test/resources/protocols/gson_user.json";
+
+        try (Writer writer = new FileWriter(filePath)) {
+            Gson gson = new GsonBuilder().setPrettyPrinting().create();
+            gson.toJson(user, writer);
+            Assert.assertTrue(Files.exists(Paths.get(filePath)));
```
Comment (KevinGilmore) [inline libraries-data-io/src/test/java/com/baeldung/serialization/protocols/GsonSerializationUnitTest.java:32]: For consistency, do a static import of the assertTrue method so you don't need to qualify it here.

File: libraries-data-io/src/test/java/com/baeldung/serialization/protocols/JacksonSerializationUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+        String filePath = "src/test/resources/protocols/jackson_user.json";
+
+        File file = new File(filePath);
+        
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.writeValue(file, user);
+
+        User deserializedUser = mapper.readValue(new File(filePath), User.class);
+        assertEquals(1, deserializedUser.getId());
```
Comment (KevinGilmore) [inline libraries-data-io/src/test/java/com/baeldung/serialization/protocols/JacksonSerializationUnitTest.java:31]: Add blank line before assertions.

**Review by KevinGilmore — CHANGES_REQUESTED** [review]
See notes regarding indentation, blank line placement, and static import.

File: libraries-data-io/pom.xml [hunk 6]
```diff
@@ -55,6 +55,52 @@
             <artifactId>docx4j</artifactId>
             <version>${docx4j.version}</version>
         </dependency>
+	<!-- gson api -->
```
Comment (KevinGilmore) [inline libraries-data-io/pom.xml:58]: The dependencies you added all are using tabs to indent.

Comment (MeenaGawande) [reply libraries-data-io/pom.xml]: My eclipse editor wasn't showing any differences before. I am able to see the changes after changing the preferences for tab spaces. I fixed it now. Please check and let me know.

**Review by KevinGilmore — APPROVED** [review]

**Pull request merged**